homeassistant/components/templated_automation/__init__.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for Templated Automation.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/templated_automation
+"""
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_SOURCE, EVENT_STATE_CHANGED, STATE_OFF, \
+    SERVICE_TURN_OFF, SERVICE_TURN_ON, ATTR_ENTITY_ID, ATTR_STATE
+
+CONF_BIDIRECTIONAL = 'bidirectional'
+CONF_INVERSE = 'inverse'
+CONF_TARGET = 'target'
+
+DEFAULT_BIDIRECTIONAL = False
+DEFAULT_INVERSE = False
+
+DOMAIN = 'templated_automation'
+
+AUTOMATION_TEMPLATE = vol.Schema({
+    vol.Required(CONF_SOURCE): cv.entity_id,
```

Comment:
Should we limit source and target entity domains to binary entity domains, eg binary_sensor, switch, light?

Reply:
I think at this point that makes sense. Ideally, I'd like to open this open so you can use a template as the source/destination, but for MVP, we should limit it.

Reply:
Sorry, I shouldn't have included lock as it won't work with turn on/off service.